system/osx_defaults.py

```diff
@@ -160,6 +172,15 @@ def _convert_type(self, type, value):
             if not isinstance(value, list):
                 raise OSXDefaultsException("Invalid value. Expected value to be an array")
             return value
+        elif type in ["dict", "dictionary"]:
+            if isinstance(value, dict):
+                return [value]
+            elif isinstance(value, list):
+                return value
+            elif isinstance(value, str):
```

Comment:
@claco In `main()` where `argument_spec` is defined, `value` is defined as:

```
            value=dict(
                default=None,
                required=False,
            )
```

Without specifying a `type` for `value`, in 2.1 this gets cast to a string.  If that `value` can accept more than just a `str`, then it needs to be updated to look like:

```
            value=dict(
                default=None,
                required=False,
                type='raw'
            )
```

The change in 2.1, is that any argument in `argument_spec` without a explicitly defined type, now is defaulted to `str`.